tox.ini

```diff
@@ -2,29 +2,29 @@
 envlist =
     flake8
     isort
-    py{38,39,310}-dj{32}-cms{310,311}
-    py{38,39,310}-dj{40}-cms{311}
-    py{39,310}-dj{40,42}-cms{41}
+    py{39,310,311}-dj{42}-cms{311}
+    py{310,311,312}-dj{42,50,51}-cms{41}
 
 skip_missing_interpreters=True
 
 [testenv]
 deps =
     -r{toxinidir}/tests/requirements/base.txt
-    dj32: Django>=3.2,<4.0
-    dj40: Django>=4.0,<4.1
     dj42: Django>=4.2,<5.0
-    cms310: django-cms>=3.10,<3.11
-    cms311: django-cms>=3.11,<3.12
+    dj50: Django>=5.0,<5.1
+    dj51: Django>=5.1,<5.2
+    cms311: django-cms>=3.11,<4
     cms41: django-cms>=4.1,<4.2
 commands =
     {envpython} --version
     {env:COMMAND:coverage} erase
-    {env:COMMAND:coverage} run setup.py test
+    {env:COMMAND:coverage} run -m pytest
     {env:COMMAND:coverage} report
 
 [testenv:flake8]
-deps = flake8
+deps =
```

Comment:
We could consider dropping everything but tests from tox. Because pre-commit and github actions can pick up these.

Then you can get the github tests to install with `uv` and install `tox` and `tox-uv`. This is a similar setup then to what I have here; https://github.com/marksweb/django-nh3/blob/main/.github/workflows/main.yml#L42

Reply:
I don't really use tox. Kept it for consistency. I'm fine with removing linting from tox. `uv` probably is for performance reasons, right?

Reply:
Tox just makes running the test suite locally easier - then you can get github to do the same for consistency.

Yeah - uv is just doing things better. _(I actually used it to setup a fresh project to show an [example](https://github.com/marksweb/django-example) of clearing cache from admin recently.)_